### PR TITLE
Network tests

### PR DESCRIPTION
**Pull request by rickramirr**
#### Code review checklist [Code Review Rubric](https://github.com/wizeline/iOS-apprenticeship-program-2020/blob/main/code-review-rubric.md)

1. DRY Principle (Don’t repeat yourself)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

2. Yagni Principle (You Aren't Gonna Need It)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

3. KISS Principle (Keep It Simple, St...d)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

4. Remove unused code

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

5. Hard-coded values

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

6. Language style guide & coding conventions
 
 - [Wizeline Style Guide](https://github.com/wizeline/mobile-documentation/blob/master/guidelines/swift-style-guide.md) 
 - [RayWenderlich Style Guide](https://github.com/raywenderlich/swift-style-guide)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

7. Architecture / Design Pattern

    - [ ] MVC  
    - [ ] MVP 
    - [ ] MVVM 
    - [ ] VIPER 

8. Architecture Implementation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

9. SOLID

- Single Responsibility Principle 

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Open Closed Principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Liskov substitutability principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Interface segregation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Dependency Inversion principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

10. Testing

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

11. Force unwrapping !

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

12. Mutable vs immutable property (Right use of `let`  and `var`)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

13. Retain cycles (Capture list)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

14. Code Documentation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

### CODE REVIEW

File: TheMovieDb/Common/UIImageView+extension.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            if let data = try? Data(contentsOf: url) {
+                if let image = UIImage(data: data) {
+                    DispatchQueue.main.async {
+                        self?.image = image
+                        completion?(image)
+                    }
+                }
+            }
+        }
+    }
```
Comment (Carlos-Barron) [inline TheMovieDb/Common/UIImageView+extension.swift]: In this case you are using two threads why?

File: TheMovieDb/Modules/Detail/DetailView.swift [hunk 2]
```diff
@@ -105,6 +103,16 @@ final class DetailView: UIViewController {
...
+        if let posterpath = viewModel.getMoviePosterPath(),
+           let posterURL = URL(string: MovieDBAPI.APIConstants.imageUrl + posterpath) {
+            if let cached = cache[posterURL.absoluteString] {
+                poster.image = cached
+            } else {
+                poster.load(url: posterURL) { [weak self] image in
+                    self?.cache[posterURL.absoluteString] = image
+                }
+            }
+        }
```
Comment (Carlos-Barron) [inline TheMovieDb/Modules/Detail/DetailView.swift]: 🤔  in this case you can handle this logic in your ViewModel and your ViewController only accepts an image

**Review by Carlos-Barron — APPROVED** [review]
Just resolve the comments

File: TheMovieDb/Common/UIImageView+extension.swift (hunk 1 above)
Comment (rickramirr) [reply TheMovieDb/Common/UIImageView+extension.swift]: I use global to request data trough the Data object and main to update the image on image view

File: TheMovieDbTests/MovieDBAPITests.swift [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                    "id": 299536,
+                    "overview": "As the Avengers...",
+                    "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
+                    "release_date": "2018-04-25",
+                    "title": "Avengers: Infinity War",
+                  }
+                ],
+                "total_pages": 792,
+                "total_results": 15831
+              }
```
Comment (Carlos-Barron) [inline TheMovieDbTests/MovieDBAPITests.swift:97]: In these cases you can make separate files and load them up in the test. If you can edit them it's perfect if don't it's ok

Comment (rickramirr) [reply TheMovieDbTests/MovieDBAPITests.swift:97]: Ok

**Pull request merged**